src/main/java/org/checkerframework/specimin/WorkDoneByUnsolvedSymbolVisitor.java

```diff
@@ -0,0 +1,56 @@
+package org.checkerframework.specimin;
+
+import java.util.Objects;
+import java.util.Set;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/** A simple class to keep track of the progress of UnsolvedSymbolVisitor. */
+public class WorkDoneByUnsolvedSymbolVisitor {
```

Comment:
It is confusing that this class' name ends in `Visitor`, but it is not a visitor. Can we come up with a name that isn't so confusing? Here are some suggestions:
* `UnsolvedSymbolVisitorProgress`
* `UnsolvedSymbolVisitorProgressTracker`
* `UnsolvedSymbolVisitorWork`